renderer/viewer/lib/chunkCacheIntegration.ts

```
@@ -0,0 +1,308 @@
+ *   - Cache hit: Server sends only a confirmation, client uses local cached packet data
+ *   - Cache miss: Server sends full map_chunk packet, client caches it for future sessions
+ * - This saves significant network bandwidth for unchanged chunks
+ *
+ * ## Server Scoping:
+ * - Memory cache is cleared when connecting to a different server (via setServerSupportsChannel)
+ * - Disk cache is server-scoped: /data/geometry-cache/{serverAddress}/ and /data/chunk-cache/{serverAddress}/
+ * - Each server has isolated cache storage to prevent data conflicts
+ */
+
+import type { MesherGeometryOutput } from './mesher/shared'
+import { computeBlockStateHash } from '../../../src/blockHash'
```

ESLint

[error] `../../../src/blockHash` import should occur before type import of `./mesher/shared`

(import/order)